Mage/src/main/java/mage/util/CardUtil.java

```diff
@@ -1292,7 +1292,19 @@ public interface SpellCastTracker {
         void addCard(Card card, Ability source, Game game);
     }
 
-    private static List<Card> getCastableComponents(Card cardToCast, FilterCard filter, Ability source, Player player, Game game, SpellCastTracker spellCastTracker, boolean playLand) {
+    /**
+     * Retrieves a list of castable components from a given card based on certain conditions.
```

Comment:
so what is a castable component? the added comment doesn't really help me understand when/why I'd want to use this function... additional explanation or example would be helpful

Reply:
We added a bit more explanation to the comment, see if it helps. But we are not the original author of that function, so it is only our interpretation.